stl/src/filesystem.cpp

```diff
@@ -796,20 +795,11 @@ namespace {
         _Stl_GetTempPath2W(_In_ DWORD nBufferLength, _Out_writes_to_opt_(nBufferLength, return +1) LPWSTR lpBuffer) {
         using _Fun_ptr = decltype(&::GetTempPath2W);
 
-        _Fun_ptr _PfGetTempPath2W;
-        {
-            static _STD atomic<_Fun_ptr> _Static{nullptr};
-
-            _PfGetTempPath2W = _Static.load(_STD memory_order_relaxed);
-            if (!_PfGetTempPath2W) {
-                const auto _Kernel32 = ::GetModuleHandleW(L"kernel32.dll");
-                _Analysis_assume_(_Kernel32);
-                _PfGetTempPath2W = reinterpret_cast<_Fun_ptr>(::GetProcAddress(_Kernel32, "GetTempPath2W"));
-                if (!_PfGetTempPath2W) {
-                    _PfGetTempPath2W = &::GetTempPathW;
-                }
-                _Static.store(_PfGetTempPath2W, _STD memory_order_relaxed); // overwriting with the same value is okay
-            }

```

Comment:
IMO we should have a transition comment here (1) so we can improve this when and if we break ABI and merge this into the DLL, and (2) so someone doesn't try to "fix" this in a year or two causing us to repeat this debugging expersience.

Reply:
Good idea, I'll add one.

Reply:
Done.